proj/src/main/webapp/test/chat.test.js

```diff
@@ -14,6 +14,77 @@ require("../app.js");
 let testHelper = require('./test-helper-functions.js');
 let chat = require('../chat.js');
 
+describe('creating and joining 1-on-1 chats', function() {
+    this.timeout(10000);
+    let testUsers = [{
+        email: 'test@test.com', 
+        password: 'test123!'
+    }, 
+    {
+        email: 'test2@test.com', 
+        password: 'test123!'
+    }];
+    let chatId;
+    const oneOnOneRef = firebase.database().ref('/chat/chats-1on1');
+
+    before(async function() {
+        for (let i = 0; i < testUsers.length; i++) {
+            await testHelper.createUserIfNotExisting(testUsers[i]);
+            await firebase.auth().signInWithEmailAndPassword(testUsers[i].email, testUsers[i].password);
+            testUsers[i]["uid"] = firebase.auth().currentUser.uid;
+            if (i == 1) {
+                chatId = oneOnOneRef.push().key;
+                await chat.createFriendChat(chatId, testUsers[0]["uid"]);
+            } else {
+                await firebase.auth().signOut();
+            }
+        }
+    });
+
+    after(async function() {
+        if (testconfig.shouldCleanUp) {
+            // Remove nodes created
+            await testHelper.emptyDatabase();
+        }
+
+        // delete created chat
+        oneOnOneRef.child(chatId).remove();
+
+        // sign out
+        await firebase.auth().signOut();
```

Comment:
I'm wondering if this should be part of the test case, but I don't have any strong feelings because it is good to make sure you are signed out when everything is finished. It might have an error if somehow the test case already signs you out though, but the same thing is true if it is in the test case and you somehow get signed out before you explicitely tell it to sign you out.